starcheck/src/lib/Ska/Parse_CM_File.pm

```diff
@@ -137,17 +153,32 @@ sub dither {
          if defined $params[$_]->{RATEP};
       $dither_period_y = 1 / ($params[$_]->{RATEY} / (2 * $pi))
          if defined $params[$_]->{RATEP};
-      push @dither, { time => $time[$_],
-		      state => $dither_state,
-		      source => $time[$_] < $bs_start ? 'history' : 'backstop',
-		      ampl_p => $dither_ampl_p,
-		      ampl_y => $dither_ampl_y,
-                      period_p => $dither_period_p,
-                      period_y => $dither_period_y,
-                      tlmsid => $params[$_]->{TLMSID},
-                  };
+      if ($dither_state eq 'DISA'){
+          # If disabled, reset the amplitudes to be 0.  The params may be nonzero onboard
+          # but we're more interested in the effective amplitudes for starcheck.
+          push @dither, { time => $time[$_],
+                          state => 'DISA',
+                          source => $time[$_] < $bs_start ? 'kadi' : 'backstop',
```

Comment:
I don't understand how `$time[$_] < $bs_start` can ever be true here.  This section of code is strictly processing backstop commands, so having this conditional here just makes me question if I understand what's going on.  In fact it did make me spend a lot more time re-reading the code, but that is not necessarily the goal.

Reply:
Good catch.  I think this was a benign error on my part when I repurposed the previous code.  Can be simplified.